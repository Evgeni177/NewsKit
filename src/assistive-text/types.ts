import {EnhancerOverrides} from '../form/types';
import {MQ} from '../utils/style';
import {WithEnhancersProps} from '../with-enhancers/types';

export type AssistiveTextProps = React.HTMLAttributes<HTMLParagraphElement> &
  Omit<
    WithEnhancersProps,
    'enhancersType' | 'isFocused' | 'componentDefaultsPath'
  > & {
    overrides?: EnhancerOverrides & {
      stylePreset?: MQ<string>;
      typographyPreset?: MQ<string>;
      minHeight?: MQ<string>;
    };
  };
