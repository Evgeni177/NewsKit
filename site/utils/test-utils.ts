import {
  renderToFragmentWithThemeFactory,
  renderWithThemeFactory,
} from 'newskit/test/test-utils';
import {docsThemeLight} from '../theme/doc-theme';

export const renderWithTheme = renderWithThemeFactory(docsThemeLight);

export const renderToFragmentWithTheme = renderToFragmentWithThemeFactory(
  docsThemeLight,
);
