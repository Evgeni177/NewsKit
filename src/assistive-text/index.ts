export * from './assistive-text';
export * from './types';
