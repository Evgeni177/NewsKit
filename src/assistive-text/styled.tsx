import {TextBlockProps} from '../text-block/types';
import {TextBlock} from '../text-block';
import {Block, BlockProps} from '../block';

import {
  getResponsiveSize,
  getTypographyPreset,
  getStylePreset,
  styled,
} from '../utils/style';

import {AssistiveTextProps} from './types';

export const StyledAssistiveTextContainer = styled(Block)<
  Omit<BlockProps, 'overrides'> &
    Pick<AssistiveTextProps, 'size' | 'children' | 'overrides'>
>`
  ${({size}) =>
    getResponsiveSize('minHeight', `assistiveText.${size}`, '', 'minHeight')};
`;

export const StyledAssistiveText = styled(TextBlock)<
  Omit<TextBlockProps, 'overrides'> & AssistiveTextProps
>`
  ${({size, state}) =>
    getStylePreset(`assistiveText.${size}`, '', {
      isInvalid: state === 'invalid',
      isDisabled: state === 'disabled',
      isValid: state === 'valid',
    })}
  ${({size}) =>
    getTypographyPreset(`assistiveText.${size}`, '', {
      withCrop: true,
    })}
`;
