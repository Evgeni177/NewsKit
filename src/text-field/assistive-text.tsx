import React from 'react';
import {StyledAssistiveText, StyledAssistiveTextContainer} from './styled';

import {AssistiveTextProps, TextFieldSize} from './types';

export const AssistiveText = ({
  overrides,
  size = TextFieldSize.Medium,
  state,
  children,
  ...props
}: AssistiveTextProps) => (
  <StyledAssistiveTextContainer size={size} overrides={overrides}>
    {children && (
      <StyledAssistiveText
        size={size}
        overrides={overrides}
        state={state}
        role={state === 'invalid' ? 'alert' : undefined}
        aria-live={state === 'invalid' ? 'polite' : undefined}
        {...props}
      >
        {children}
      </StyledAssistiveText>
    )}
  </StyledAssistiveTextContainer>
);
