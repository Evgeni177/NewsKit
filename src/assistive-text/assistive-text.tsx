import React from 'react';
import {TextFieldSize} from '../text-field';
import {StyledAssistiveText, StyledAssistiveTextContainer} from './styled';

import {AssistiveTextProps} from './types';

export const AssistiveText = ({
  overrides,
  size = 'medium' as TextFieldSize,
  state,
  children,
  ...props
}: AssistiveTextProps) => (
  <StyledAssistiveTextContainer size={size} overrides={overrides}>
    {children && (
      <StyledAssistiveText
        aria-disabled={state === 'disabled' ? true : undefined}
        size={size}
        overrides={overrides}
        state={state}
        role={state === 'invalid' ? 'alert' : undefined}
        aria-live={state === 'invalid' ? 'polite' : undefined}
        {...props}
      >
        {children}
      </StyledAssistiveText>
    )}
  </StyledAssistiveTextContainer>
);
