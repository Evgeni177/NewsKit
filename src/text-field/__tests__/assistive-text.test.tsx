import React from 'react';
import {AssistiveText} from '../assistive-text';
import {AssistiveTextProps, TextFieldSize} from '..';
import {renderToFragmentWithTheme} from '../../test/test-utils';

const renderValidAssistiveText = () => (
  <>
    <AssistiveText state="invalid">AssistiveText</AssistiveText>
  </>
);

describe('Assistive Text', () => {
  [TextFieldSize.Small, TextFieldSize.Medium, TextFieldSize.Large].forEach(
    size => {
      test(`renders ${size} AssistiveText`, () => {
        const props: AssistiveTextProps = {
          children: 'Assistive Text',
          size,
        };
        const fragment = renderToFragmentWithTheme(AssistiveText, props);
        expect(fragment).toMatchSnapshot();
      });
    },
    test('should render invalid Assistive Text', () => {
      const props: AssistiveTextProps = {
        children: 'Assistive Text',
        state: 'invalid',
      };

      const fragment = renderToFragmentWithTheme(
        renderValidAssistiveText,
        props,
      );
      expect(fragment).toMatchSnapshot();
    }),
  );
});
