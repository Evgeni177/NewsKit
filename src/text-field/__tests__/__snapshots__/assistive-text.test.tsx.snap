// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Assistive Text renders large AssistiveText 1`] = `
<DocumentFragment>
  .emotion-0 {
  min-height: 12px;
}

.emotion-1 {
  margin: 0;
  padding: 1px 0;
  color: #535353;
  font-family: "Noto Sans",sans-serif;
  font-size: 16px;
  line-height: 1.5;
  font-weight: 500;
  letter-spacing: 0;
}

.emotion-1:disabled {
  color: #c0c0c0;
}

.emotion-1::before {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-bottom: -0.36250000000000004em;
}

.emotion-1::after {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-top: -0.4625em;
}

<div
    class="emotion-0"
  >
    <p
      class="emotion-1"
    >
      Assistive Text
    </p>
  </div>
</DocumentFragment>
`;

exports[`Assistive Text renders medium AssistiveText 1`] = `
<DocumentFragment>
  .emotion-0 {
  min-height: 12px;
}

.emotion-1 {
  margin: 0;
  padding: 1px 0;
  color: #535353;
  font-family: "Noto Sans",sans-serif;
  font-size: 14px;
  line-height: 1.4285714285714286;
  font-weight: 500;
  letter-spacing: 0;
}

.emotion-1:disabled {
  color: #c0c0c0;
}

.emotion-1::before {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-bottom: -0.32678571428571435em;
}

.emotion-1::after {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-top: -0.4267857142857143em;
}

<div
    class="emotion-0"
  >
    <p
      class="emotion-1"
    >
      Assistive Text
    </p>
  </div>
</DocumentFragment>
`;

exports[`Assistive Text renders small AssistiveText 1`] = `
<DocumentFragment>
  .emotion-0 {
  min-height: 12px;
}

.emotion-1 {
  margin: 0;
  padding: 1px 0;
  color: #535353;
  font-family: "Noto Sans",sans-serif;
  font-size: 14px;
  line-height: 1.4285714285714286;
  font-weight: 500;
  letter-spacing: 0;
}

.emotion-1:disabled {
  color: #c0c0c0;
}

.emotion-1::before {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-bottom: -0.32678571428571435em;
}

.emotion-1::after {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-top: -0.4267857142857143em;
}

<div
    class="emotion-0"
  >
    <p
      class="emotion-1"
    >
      Assistive Text
    </p>
  </div>
</DocumentFragment>
`;

exports[`Assistive Text should render invalid Assistive Text 1`] = `
<DocumentFragment>
  .emotion-0 {
  min-height: 12px;
}

.emotion-1 {
  margin: 0;
  padding: 1px 0;
  color: #d20600;
  font-family: "Noto Sans",sans-serif;
  font-size: 14px;
  line-height: 1.4285714285714286;
  font-weight: 500;
  letter-spacing: 0;
}

.emotion-1::before {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-bottom: -0.32678571428571435em;
}

.emotion-1::after {
  content: '';
  display: block;
  height: 0;
  width: 0;
  margin-top: -0.4267857142857143em;
}

<div
    class="emotion-0"
  >
    <p
      aria-live="polite"
      class="emotion-1"
      role="alert"
    >
      AssistiveText
    </p>
  </div>
</DocumentFragment>
`;
